### PR TITLE
docs: Improve development guide.

### LINKED ISSUE #341: Improve development guide
We have this [guide](https://github.com/zulip/zulip-electron/blob/master/development.md) but it's not detailed. It would be great to have a detailed guide on how to install the dev env on various OS Windows/macOS/Linux which will include the instructions, screenshots, troubleshooting etc.

### PR DESCRIPTION
**Pull request by vsvipul**
Minifies the current development guide by
a significant amount. Also, removes system requirements
as no credible source for them. Removes dependency
specific instructions and adds links to instructions
to install them.

Fixes #341.

Preview [here](https://github.com/zulip/zulip-desktop/blob/e66524e80fc49c51ef8c21703b533dbb0515b35c/development.md).

**You have tested this PR on:**
  - [x] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

**Comment by vsvipul** [comment]
@akashnimare @abhigyank please review.

File: development.md [hunk 1]
```diff
@@ -1,152 +1,116 @@
...
-## MacOS
+[node-mac]: https://nodejs.org/en/download/package-manager/#macos
 
+If [NPM](https://www.npmjs.com/get-npm) and [node-gyp](https://github.com/nodejs/node-gyp#installation) don't come bundled with your Node.js installation, you will need to install them manually. 
 
-  **Node JS**
-  Go to the [Node.js Downloads page](https://nodejs.org/en/download/). Download Node.js for MacOS (`v6.9.0` or above recommended). Run the downloaded Node.js `.pkg` Installer. You're finished! To ensure Node.js has been installed, run `node -v` in your terminal - you should get something like `v6.9.0` or above
+### Windows
 
+- On Windows, your C++ compiler should be Visual Studio 2015 or later.
```
Comment (priyank-p) [inline development.md]: The script below `npm install --global windows-build-tools` does this. Also I think for windows section first instruction should be install node, then this.

File: development.md [hunk 2]
```diff
@@ -1,152 +1,116 @@
...
+- Use Git Bash and always run it as administrator.
```
Comment (priyank-p) [inline development.md]: Why is this required?

**Review by priyank-p — COMMENTED** [review]
I left a few comments for windows section

File: development.md (hunk 1 above)
Comment (priyank-p) [reply development.md]: (To be clear first install node instruction, then the `npm install --global windows-build-tools`)

File: development.md (hunk 2 above)
Comment (vsvipul) [reply development.md]: @priyank-p read about it somewhere in zulip docs. I think [here](https://zulip.readthedocs.io/en/latest/development/setup-vagrant.html). Should i remove it ?

File: development.md (hunk 1 above)
Comment (vsvipul) [reply development.md]: @priyank-p  will do.

File: development.md (hunk 2 above)
Comment (priyank-p) [reply development.md]: Yeah, that is specifically for zulip development environment not necessary for this project.

Comment (vsvipul) [reply development.md]: Done.

File: development.md [hunk 3]
```diff
@@ -1,152 +1,114 @@
...
-3. **Python (v2.7.x recommended)**
+To package the app into an installer:
+```
+npm run dist
+```
 
-	`$ sudo apt install python2.7`
+This command will produce distributable packages or installers for the
+operating system you're running on:
+* on Windows, a Windows installer file
```
Comment (kanishk98) [inline development.md]: We should probably also mention the `nsis.7z` and WebSetup `.exe` file extensions here, because the packaging concept can be a little confusing to new developers.

Comment (vsvipul) [reply development.md]: done.

File: development.md [hunk 4]
```diff
@@ -1,152 +1,114 @@
...
-
-Now you are ready for next step [: Get Zulip Desktop Code.](https://github.com/zulip/zulip-desktop/blob/master/development.md#get-zulip-desktop-code)
-
-
-## Windows
-
-  **Node JS**
-  Go to the [Node.js Downloads page](https://nodejs.org/en/download/). Download Node.js for windows (`v6.9.0` or above recommended). Run the downloaded Node.js `.msi` Installer. You're finished! To ensure Node.js has been installed, run `node -v` in your terminal - you should get something like `v6.9.0` or above
-
-
```
Comment (akashnimare) [inline development.md:82]: @vsvipul I think we should keep it?

**Review by akashnimare — CHANGES_REQUESTED** [review]

File: development.md (hunk 4 above)
Comment (vsvipul) [reply development.md:82]: @akashnimare i have kept this instruction in line 48.

**Review by akashnimare — APPROVED** [review]

**Review by abhigyank — APPROVED** [review]

**Comment by akashnimare** [comment]
Thanks, @vsvipul.
